api/types.proto

```diff
@@ -624,6 +624,15 @@ message RaftConfig {
 	uint32 election_tick = 5;
 }
 
+
+// SecretsConfig specifies cluster-level secrets settings.
+message SecretsConfig {
+	// MaxSecrets specifies how many user-specified secrets there can be
+	uint32 max_secrets = 1;
+	// MaxSecretVersions specifies how many versions of a secret there can be
+	uint32 max_secret_versions = 2;
```

Comment:
No, as far as I know protocol buffers doesn't have the concept of a default value. The code that interprets this configuration will have to check for `0` and handle that as a special case.